Library/Homebrew/os/linux/elf.rb

```diff
@@ -78,6 +79,18 @@ def dynamic_elf?
     end
   end
 
+  def corrupted_elf?
+    # elf files that throw the following message when parsed with readelf:
+    # Warning: possibly corrupt ELF header
+    # it has a non-zero program header offset, but no program headers
+    # Example: the early-section-headers.so unittest file from the rust package
+    message = ""
+    Open3.popen3("readelf", "-d", to_path) do |_, _, stderr|
+      message = stderr.read
+    end
+    message.include?("readelf: Warning: possibly corrupt ELF header")
```

Comment:
```suggestion
    message.start_with?("readelf: Warning: possibly corrupt ELF header")
```
or `==` if that's the complete message.

Reply:
The message is a little bit longer. I will go for start_with.